packages/medusa/src/strategies/price-selection.ts

```diff
@@ -7,15 +7,20 @@ import {
 } from "../interfaces/price-selection-strategy"
 import { MoneyAmountRepository } from "../repositories/money-amount"
 import { EntityManager } from "typeorm"
+import { FlagRouter } from "../utils/flag-router"
+import TaxInclusivePricingFeatureFlag from "../loaders/feature-flags/tax-inclusive-pricing"
+import { TaxServiceRate } from "../types/tax-service"
 
 class PriceSelectionStrategy extends AbstractPriceSelectionStrategy {
   private moneyAmountRepository_: typeof MoneyAmountRepository
+  private featureFlagRouter_: FlagRouter
   private manager_: EntityManager
 
-  constructor({ manager, moneyAmountRepository }) {
+  constructor({ manager, featureFlagRouter, moneyAmountRepository }) {
     super()
     this.manager_ = manager
     this.moneyAmountRepository_ = moneyAmountRepository
```

Comment:
thought(non-blocking): we need to remove this dependency at some point, and move all necessary functionality into the service layer (maybe PricingService)